pkgs/development/libraries/glibc/check-etc-zoneinfo.patch

```diff
@@ -0,0 +1,24 @@
+diff --git a/time/tzfile.c b/time/tzfile.c
+index 040a5e34..e7c344cb 100644
+--- a/time/tzfile.c
++++ b/time/tzfile.c
+@@ -146,7 +146,18 @@ __tzfile_read (const char *file, size_t extra, char **extrap)
+ 	tzdir = default_tzdir;
+       if (__asprintf (&new, "%s/%s", tzdir, file) == -1)
+ 	goto ret_free_transitions;
+-      file = new;
++
++      // nixos specific adjustment - also check /etc/zoneinfo. Does an extra syscall
++      // to __stat64 which is not amazing. Not sure how else to solve. Might be worth
++      // checking "/etc/zoneinfo/" first in case most people use this on nixos itself.
++      struct stat64 st;
++      if (__stat64 (new, &st) == 0) {
```

Comment:
I spent a little bit more time thinking about this. I think my current logic is broken because it always defaults to `/etc/zoneinfo`, even if `TZDIR` points to somewhere that would have failed with `ENOENT`. I changed the logic to only check `/etc/zoneinfo` if `TZDIR` is is `TZDEFAULT`.

I also don't think we can drop the `__stat64` further down because there are several other paths through the code that rely on it. And now that I've changed the check to `ENOENT` the newly added `__stat64` would not even be checking the same error condiions.